File: Kotlin/1047.kt
import java.util.Scanner

fun main() {
    val input = Scanner(System.`in`);
    val t_minutos_inicio = input.nextInt() * 60 + input.nextInt() ;
    val t_minutos_fin = input.nextInt() * 60 + input.nextInt() ;
    val t = if(t_minutos_fin > t_minutos_inicio) t_minutos_fin - t_minutos_inicio 
    else 1440 - t_minutos_inicio + t_minutos_fin;
    val h = t / 60;
    val m = t % 60;
   
    println("O JOGO DUROU $h HORA(S) E $m MINUTO(S)");
}
    

File: Php/1047.php
<?php
[$h_inicio, $m_inicio, $h_fin, $m_fin] = explode(" ", readline());
$t_minutos_inicio = 60 * $h_inicio + $m_inicio;
$t_minutos_fin = 60 * $h_fin + $m_fin;

$t = $t_minutos_fin > $t_minutos_inicio ? 
$t_minutos_fin - $t_minutos_inicio : 
1440 - $t_minutos_inicio + $t_minutos_fin;

$h = intdiv($t, 60);
$m = $t % 60;

printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)\n" ,$h, $m)

?>


